### DISCUSSION TITLE
Does cy.intercept() work with only API automation?

### DISCUSSION THREAD

**Discussion by pratikpatel1221** [cypress-io/cypress#14942] [Questions and Help] [answered]
It is not working for me with below code snippet

```
 it('testing', () => {

        cy.intercept('GET', 'https://reqres.in/api/users', { fixture: 'userData.json' }).as('users')
        cy.request('GET','https://reqres.in/api/users').then((res) => {
            cy.log(res.body)
              })

        cy.wait('@users').then((interception) => {
                cy.log(interception)
        })
    })
```

**Comment by jennifer-shehane** [comment, marked as answer]
cy.intercept() only sees requests made by the browser (your app under test). For pure API testing with no UI, use cy.request() to call endpoints directly and assert on the response, no interception needed. cy.intercept is for stubbing/spying the app’s own XHR/fetch, not for standalone API calls.
